tests/functional/tests/flush/flush_test.py

```diff
@@ -0,0 +1,289 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+import math
+import logging
+import random
+from ctypes import c_int
+
+from pyocf.types.core import Core
+from pyocf.types.volume import Volume
+from pyocf.types.data import Data
+from pyocf.types.io import IoDir
+from pyocf.types.shared import OcfCompletion
+from pyocf.types.cache import CacheMode, CacheLineSize, Cache
+from pyocf.utils import Size as Size
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def run_io(device, queue, data, direction):
+    write_data = Data.from_bytes(data)
+    io = device.new_io()
+    io.set_data(write_data)
+    io.configure(0, write_data.size, direction, 0, 0)
+    io.set_queue(queue)
+
+    cmpl = OcfCompletion([("err", c_int)])
+    io.callback = cmpl.callback
+    io.submit()
+    cmpl.wait()
+    assert cmpl.results["err"] == 0
+
+
+def check_configuration(cache, expected_conf):
+    stats = cache.get_stats()
+    assert stats["conf"]["cache_line_size"] == expected_conf["cache_line_size"]
+    assert stats["conf"]["cache_mode"] == expected_conf["cache_mode"]
+
+
+def check_statistics(stats, expected_occupancy, expected_dirty, expected_clean):
+    assert stats["usage"]["occupancy"]["value"] == expected_occupancy
+    assert stats["usage"]["dirty"]["value"] == expected_dirty
+    assert stats["usage"]["clean"]["value"] == expected_clean
+
+
+# TODO: remove xfail when wb cache mode will be fixed
+@pytest.mark.xfail
+@pytest.mark.parametrize("cls", CacheLineSize)
+def test_wb_flush_cache(pyocf_ctx, start_cache_with_core, cls):
+    cache_device, core_device, cache, core = start_cache_with_core(Size.from_MiB(30),
+                                                                   Size.from_MiB(60),
+                                                                   cache_mode=CacheMode.WB,
+                                                                   cache_line_size=cls)
+
+    check_configuration(cache, {"cache_line_size": cls, "cache_mode": CacheMode.WB})
```

Comment:
Do we need to check if the cache we just started has the configuration we wanted? I think it should be already tested in start/stop tests and here it's only noise.
Same for all the other tests

Reply:
Done